### PR TITLE
report: add new command

### PR DESCRIPTION
**Pull request by watilde**
Hi, I added report command for fixing #5214. That proposal might make support useful, people become needn't bring manually a npm-debug.log to the issue description in GitHub.

Well, I'd like to use `gist` but I think it might makes npm a little bit complex so I've built in this way.
See also: https://github.com/npm/npm/issues/5214#issuecomment-77362757 

Thanks.

### CODE REVIEW

**Comment by rlidwka** [comment]
> Well, I'd like to use gist but I think it might makes npm a little bit complex so I've built in this way.

Just post it to some kind of pastebin with 1 month expiration time imho.

**Comment by othiym23** [comment]
I like this! I do think it's going to take a little effort to land this, but now is probably as good a time as ever to get this moving along. This is an actual honest-to-gosh feature in npm, and as such needs a little extra effort to get right:

- [ ] document the command as it currently stands
- [ ] land #6744
- [ ] land #7614
- [ ] figure out how to hang on to debug logs (putting them somewhere ephemeral prevents us from being able to refer back to specific support cases later, and also limits our ability to do data mining and learn from our support cases, so is a no go – maybe npm will write a microservice to handle this)

It would be fantastic to actually implement #5214, as we've been wanting this for nearly a year, so I'm willing to budget some of npm's time towards making sure this keeps moving forward. Thanks, @watilde!

**Comment by smikes** [comment]
What happens on issues/new if the user is not logged in to github?

**Comment by watilde** [comment]
Sorry for my late reply,

@rlidwka Thanks for your information! I'll use it a little, but maybe I can't decide to include that to this patch in this time..

@othiym23 Sounds great! I'm going to added a document in `doc/api` | `doc/cli` to this patch.

@smikes Thanks for your comment, Redirect to [this URL](https://github.com/login?return_to=https%3A%2F%2Fgithub.com%2Fnpm%2Fsupport-cli%2Fissues%2Fnew%3Fbody%3Dblahblah), after logged in report process will go well.

**Comment by othiym23** [comment]
1 down, 3 to go! Thanks for the docs!

By the way, I think this will probably be something that lands after the release of `npm@3`, as I'm trying to hold off on landing any more significantly-sized features to the `npm@2` code base that aren't directly related to support for the new npm registry infrastructure. So you may want to rebase against `multi-stage` at some point, and also be prepared to wait a little while for this one to get merged. But getting this in will be great!

**Comment by watilde** [comment]
@othiym23 I agreed to be something that lands after the release of npm@3. I'll git rebase `multi-stage` and wait as a `next-major`! I hope to make it helps npm supports, thanks.

**Comment by othiym23** [comment]
This has been moved to the [npm roadmap](https://github.com/npm/npm/wiki/Roadmap), which we're using instead of the confusing `next-*` labels now. Also, because this PR is pretty old and would need some work to land against master, I'm going to close this particular PR.

@watilde -san, once `npm@3` has landed, would you be interested in working with us to get a new version of this put together that includes #7611 and #6350? npm, Inc. now has an actual (and fantastic) support team to help out with the issues on this repo, but it would be great if we could channel front-line support into a more useful place.

File: lib/report.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  var body = ""
+  var exist = fs.existsSync(npmDebugLog)
+  var opt = ""
+
+  if (exist || title) opt += "?"
+  if (exist) {
+    body = "npm-debug.log:\n"
+    body += "```\n"
+    body += fs.readFileSync(npmDebugLog)
+    body += "```\n"
```
Comment (legodude17) [inline lib/report.js]: I think that it should ask user for input like:
`What were you trying to do?`
`Any other comments?`
Stuff like that. Also, maybe write to a file and then let the user edit it before sending it off?

**Pull request closed without merging**